### ISSUE TITLE
[auth] rework tokens

### ISSUE THREAD

**Issue by Subroutine7901** [jerimiah-smiggins/smiggins#101]
Pretty much I talked to a guy who's made actual big company auth systems and he said that having tokens that expire is actually a good idea

**Comment by trinkey** [comment]
makes sense, would need to be communicated to frontend though or just force relogin which is kinda stupid though

**Comment by Subroutine7901** [comment]
I think he mentioned relogging in being how they did it, maybe make them expire every few days? We need to do research.

**Comment by trinkey** [comment]
we could do a discord and just have it reset on password or username change, but also we should probably move to a better hashing algorithm with more output bytes

**Comment by Subroutine7901** [comment]
Yeah, wait discord does that?

**Comment by trinkey** [comment]
your discord token only changes when you change your password i believe

**Comment by Subroutine7901** [comment]
So aparently logging out invalidates it. Also tokens are per session, so uhhh
 https://www.reddit.com/r/discordapp/comments/vrotde/how_do_tokens_work/

**Comment by trinkey** [comment]
ok

**Comment by trinkey** [comment]
gonna rename this to "rework tokens" because we kinda need to

**Comment by Subroutine7901** [comment]
Ok uh with this

> makes sense, would need to be communicated to frontend though or just force relogin which is kinda stupid though

we have the frontend delete the invalidated account in the account switcher and do the same thing as adding a new account for the account switcher, with the username maybe filled in already??? also a prompt or modal for it would be good too

**Comment by trinkey** [comment]
yknow we could just use django to manage accounts when this happens

**Comment by Subroutine7901** [comment]
Proceeds to never happen

**Comment by trinkey** [comment]
django's default session system kinda sucks ngl i did make my own for https://git.trinkey.com/t/auth so we could use that maybe if this ever happens

**Comment by trinkey** [comment]
lowkey might do this soon. shouldn't be too hard to make some migration thing

```py
password_hash: str # hashed password, using django
token: str # randomly generated token, independent from password. gets reset when changing password
legacy_token: str # the legacy token, stored only for migration purposes. user would need to re-login to their account in order to  complete migration, would be set to null afterwards
```
